File: examples/equos-nodejs-integration/src/run-conversation.ts
import { client } from "./client.js";

const characterId = process.env.EQUOS_CHARACTER_ID;
if (!characterId) throw new Error("EQUOS_CHARACTER_ID is not set");

const { conversation, consumerAccessToken } =
  await client.conversations.startConversation({
    createEquosConversationRequest: {
      name: "Demo conversation",
      characterId,
      consumer: { name: "Demo User", identity: "demo-user" },
    },
  });

console.log("Conversation:", conversation.id, conversation.status);
console.log("Server URL:", conversation.serverUrl);
console.log("Consumer access token:", consumerAccessToken);
console.log(
  "Hand these to the browser / React SDK to connect. Sleeping 5s then stopping…",
);

await new Promise((resolve) => setTimeout(resolve, 5_000));

const stopped = await client.conversations.stopConversation({
  id: conversation.id,
});
console.log("Stopped. Status:", stopped.status);

File: examples/equos-python-integration/src/python_integration/run_conversation.py
import os
import time

from equos.models import CreateEquosConversationRequest, EquosParticipantIdentity

from python_integration.client import get_client


def run_conversation() -> None:
    client = get_client()

    character_id = os.getenv("EQUOS_CHARACTER_ID")
    if not character_id:
        raise RuntimeError("EQUOS_CHARACTER_ID is not set")

    response = client.conversations.start(
        CreateEquosConversationRequest(
            name="Demo conversation",
            character_id=character_id,
            consumer=EquosParticipantIdentity(name="Demo User", identity="demo-user"),
        )
    )

    print("Conversation:", response.conversation.id, response.conversation.status)
    print("Server URL:", response.conversation.server_url)
    print("Consumer token:", response.consumer_access_token)
    print("Sleeping 10s then stopping…")
    time.sleep(10)

    stopped = client.conversations.stop(response.conversation.id)
    if stopped:
        print("Stopped. Status:", stopped.status)


if __name__ == "__main__":
    run_conversation()
